my-code/src/pages/Home/index.js

```diff
@@ -0,0 +1,124 @@
+import LikeButton from '../../components/LikeButton';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import history from '../../services/history';
+import { fetchData } from '../../services/api';
+import { Container, MoviesGrid } from './styles';
+
+export default function Home() {
+  const dispatch = useDispatch();
+
+  // States
+  const [isLoading, setIsLoading] = useState(false);
+
+  // Redux state: get the movies list.
+  const movies = useSelector(state => {
+    return state.movieReducer;
+  });
+
+  /**
+   * Fetches data from the API.
+   * @param {integer} currentPage
+   */
+  async function getData(currentPage) {
+    setIsLoading(true);
+    const searchResult = await fetchData(movies.searchStr, currentPage, 10);
+
+    dispatch({
+      type: 'ADD_MOVIES',
+      movies: searchResult,
+    });
+
+    setIsLoading(false);
+  }
+
+  /**
+   * Handles the click on a movie poster.
+   * Loads the Movie page.
+   * @param {*} id
+   */
+  function handleMovieClick(id) {
+    history.push(`/movie?imdbID=${id}`);
+  }
+
+  /**
+   * Handles the click on a page number.
+   * @param {*} Object
+   */
+  function handlePageClick({ selected }) {
+    const newPage = parseInt(selected, 10) + 1;
+    getData(newPage);
+  }
+
+  return (
+    <Container>
+      <Search />
+
+      {isLoading && <LoadingIndicator text="Searching movies database..." />}
+
+      {!isLoading && (
+        <MoviesGrid className="p-grid p-justify-center">
+          {movies.list.map((movie, index) => (
+            <div className="p-col-12 p-md-5 p-lg-4 p-xl-3 " key={movie.imdbID}>
+              <div
+                className="movie-wrapper"
+                onClick={() => handleMovieClick(movie.imdbID)}
+                onKeyDown={() => handleMovieClick(movie.imdbID)}
+                role="button"
+                tabIndex={index}>
+                {movie.Poster !== 'N/A' && (
+                  <ImageFadeIn
+                    className="movie-poster"
+                    src={movie.Poster}
+                    loadAsBackgroundImage
+                    opacityTransition={1}
+                  />
+                )}
+                {movie.Poster === 'N/A' && (
+                  <div className="movie-no-poster">
+                    {movie.Title} <br />
+                    <span className="poster-not-found">(No poster)</span>
+                  </div>
+                )}
+
+                <div className="movie-btn-like">
+                  <LikeButton imdbID={movie.imdbID} />
+                </div>
+
+                <div className="movie-info">
+                  <span>
+                    {movie.Title}
+                    <br />
+                    <small>{movie.Year}</small>
+                  </span>
+                </div>
+              </div>
+            </div>
+          ))}
+        </MoviesGrid>
+      )}
+
+      {!isLoading && movies.pageCount > 0 && (
+        <ReactPaginate
```

Comment:
It would be nice if the pagination updates the browser history.

Reply:
What are the advantages of doing this since the last page visited is stored on the redux store/localStorage?